src/common/proc.c

```diff
@@ -177,7 +177,8 @@ crt_proc_daos_prop_t(crt_proc_t proc, crt_proc_op_t proc_op, daos_prop_t **data)
 		prop->dpp_reserv = tmp;
 		rc = crt_proc_prop_entries(proc, proc_op, prop);
 		if (rc) {
-			daos_prop_free(prop);
+			D_FREE(prop->dpp_entries);
```

Comment:
I would suggest to free all internal structures also.
```suggestion
			crt_proc_prop_entries(proc, CRT_PROC_FREE, prop);
			D_FREE(prop->dpp_entries);
```

Reply:
It's in crt_proc_prop_entries() that the crash occurs!

Reply:
ok. Thanks.

Reply:
There are lots of other minor fixes and cleanups in that PR, this is just to fix the one that has the potential to crash the client, rather than leak resources etc.

Reply:
if crt_proc_prop_entries() failed in the middle (allocated paritial entries), then D_FREE(prop->dpp_entries) possibly cause mem leak.
looks like dmitry's suggestion is good.